src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,103 @@
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+
+/**
+ * Edits a person in the address book.
+ */
+public class EditCommand extends Command{
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits a person in the address book. "
+            + "Contact details can be re-marked private by prepending 'p' to the prefix or not.\n"
+            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Example: " + COMMAND_WORD
+            + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+
+    public static final String MESSAGE_SUCCESS = "Person edited: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+
+    private final Person toEdit;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(String name,
+                      String phone, boolean isPhonePrivate,
+                      String email, boolean isEmailPrivate,
+                      String address, boolean isAddressPrivate,
+                      Set<String> tags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toEdit = new Person(
+                new Name(name),
+                new Phone(phone, isPhonePrivate),
+                new Email(email, isEmailPrivate),
+                new Address(address, isAddressPrivate),
+                new UniqueTagList(tagSet)
+        );
+    }
+
+    public EditCommand(Person toAdd) {
+        this.toEdit = toAdd;
+    }
+
+    public ReadOnlyPerson getPerson() {
+        return toEdit;
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson toDelete = getPersonsWithName(toEdit.getName().toString());
+            if (toDelete != null) {
+                addressBook.removePerson(toDelete);
+            } else {
+                return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+            }
+            addressBook.addPerson(toEdit);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toEdit));
+        } catch (UniquePersonList.DuplicatePersonException dpe) {
+            return new CommandResult(MESSAGE_DUPLICATE_PERSON);
+        } catch (UniquePersonList.PersonNotFoundException pnfe) {
+            return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
```

Comment:
Good implementation of the edit person. Duplicacy and missing cases are handled.